### DISCUSSION TITLE
In the DEM example, how do I estimate hashgrid size?

### DISCUSSION THREAD

**Discussion by jlinBE** [NVIDIA/warp#29] [Q&A] [answered]
From the DEM column collapse example:

        self.grid = wp.HashGrid(128, 128, 128, self.device)
        self.grid_cell_size = self.point_radius*5.0

This is used for a 32 by 128 by 32 column with radius of 0.1 m.

My question is, if I want to 

1. increase the size of the column, or 
2. have multiple columns within the same simulation 

how do I estimate how large a hashgrid I would need, to avoid crashing or slowdowns?  Is it dependent solely on particle count?  Does it matter how spread-out the particles are?

**Comment by mmacklin** [comment, marked as answer]
Larger hash-grids reduce the number of hash collisions, i.e.: particles are more likely to be hashed to cells that only contain particles near by them. 128^3 = 2M cells, so that would be the maximum number of particles you could fit without any collisions (assuming they were all perfectly distributed at cell centers).

You should try and also size your grid roughly to the layout of the points e.g.: if you had a very sheet of particles along the x-y plane then you don't need many cells in the z direction. Likewise, the worst case is when all the points lie in a single hash-grid cell, hash grids work best when the particles are spaced roughly at the same length as the cell size / radius (passed to the `HashGrid.build()` method).

Cheers,
Miles
